frappe/public/js/frappe/ui/filters/filter.js

```diff
@@ -111,7 +111,7 @@ frappe.ui.Filter = class {
 			let fieldtype = null;
 
 			if (["in", "like", "not in", "not like"].includes(condition)) {
-				fieldtype = "Data";
+				fieldtype = "Link";
```

Comment:
This field needs to be multiselect for in/not in filters. Options need to be set dynamically based on last input. Look at Communication.js for example. 

https://github.com/frappe/frappe/blob/ecd37297ef93083cbb6024caff63fd98f7ba77c2/frappe/public/js/frappe/views/communication.js#L195-L210